src/components/safe-apps/SafeAppLandingPage/AppActions.tsx

```diff
@@ -111,3 +135,45 @@ const AppActions = ({ wallet, onConnectWallet, safes, chain, appUrl }: Props): R
 }
 
 export { AppActions }
+
+const getCompatibleSafes = (
+  ownedSafes: { [chainId: string]: string[] },
+  compatibleChains: string[],
+  chainsData: ChainInfo[],
+): CompatibleSafesType[] => {
+  return compatibleChains.reduce<CompatibleSafesType[]>((safes, chainId) => {
+    const chainData = chainsData.find((chain: ChainInfo) => chain.chainId === chainId)
+
+    return [
+      ...safes,
+      ...(ownedSafes[chainId] || []).map((address) => ({
+        address,
+        chainId,
+        shortName: chainData?.shortName,
+      })),
+    ]
+  }, [])
+}
+
+const getDefaultSafe = (
+  compatibleSafes: CompatibleSafesType[],
+  chainId: string,
+  lastUsedSafe = '',
+): CompatibleSafesType => {
+  // as a first option, we use the last used Safe in the provided chain
+  const lastViewedSafe = compatibleSafes.find((safe) => safe.address === parsePrefixedAddress(lastUsedSafe).address)
+
+  if (lastViewedSafe) {
+    return lastViewedSafe
+  }
+
+  // as a second option, we use any user Safe in the provided chain
+  const safeInTheSameChain = compatibleSafes.find((safe) => safe.chainId === chainId)
+
+  if (safeInTheSameChain) {
+    return safeInTheSameChain
+  }
+
+  // as a fallback we salect a random compatible user Safe
+  return compatibleSafes[0]
```

Comment:
Why do we fall back to arbitrary choices? Looking at the original [epic](https://github.com/safe-global/safe-pm/issues/60) it clearly defines what should be the pre-selected value:
> User with at least one Safe defined in this Chain
If the user has defined at least a safe in the safe app chain we display:

> A selector with all the safes compatibles with the safe app (**the last safe used as a default selected option** if it is present).
A button to open the safe app with a safe demo.

Reply:
Also the current ticket says that the expected outcome should be the last used Safe:
>Expected result
Last connected safe should be detected and be auto selected for the user.

Reply:
I think that we can keep the same behavior that we have in the legacy codebase to facilitate QA work.

In the future we can create an improvement to change this.

Reply:
Could you please document this in the ticket for future reference? So if someone revisits this later they can track down the change. At the moment, all tickets say that it should select the last used safe.

Reply:
Also, I'm unsure if including additional behaviour facilitates QA's work. At first glance, I'd say the QA has more cases to test versus a single case with the last used safe

Reply:
I updated[ the epic](https://github.com/safe-global/safe-pm/issues/60) for future reference:

![Captura de pantalla 2022-10-06 a las 10 38 24](https://user-images.githubusercontent.com/26763673/194264476-24db67bc-dc0c-43bc-8f6f-300418d5d5d4.png)

Reply:
gracias señor